process/smartContract/processorV2/process_test.go

```diff
@@ -0,0 +1,4040 @@
+
+	scr := &smartContractResult.SmartContractResult{Value: big.NewInt(1)}
+	assert.False(t, sc.isInformativeTxHandler(scr))
+
+	scr.Value = big.NewInt(0)
+	scr.CallType = vmData.AsynchronousCallBack
+	assert.False(t, sc.isInformativeTxHandler(scr))
+
+	scr.CallType = vmData.DirectCall
+	scr.Data = []byte("@abab")
+	assert.True(t, sc.isInformativeTxHandler(scr))
+
+	scr.Data = []byte("ab@ab")
+	scr.RcvAddr = make([]byte, 32)
+	assert.False(t, sc.isInformativeTxHandler(scr))
+
+	scr.RcvAddr = []byte("address")
+	assert.True(t, sc.isInformativeTxHandler(scr))
+
+	_ = builtInFuncs.Add("ab", &mock.BuiltInFunctionStub{})
+	assert.False(t, sc.isInformativeTxHandler(scr))
+}
+
+func TestCleanInformativeOnlySCRs(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockSmartContractProcessorArguments()
+	builtInFuncs := builtInFunctions.NewBuiltInFunctionContainer()
+	arguments.BuiltInFunctions = builtInFuncs
+	arguments.ArgsParser = smartContract.NewArgumentParser()
+	sc, _ := NewSmartContractProcessorV2(arguments)
+
+	scrs := make([]data.TransactionHandler, 0)
+	scrs = append(scrs, &smartContractResult.SmartContractResult{Value: big.NewInt(1)})
+	scrs = append(scrs, &smartContractResult.SmartContractResult{Value: big.NewInt(0), Data: []byte("@6b6f")})
+
+	finalSCRs, logs := sc.cleanInformativeOnlySCRs(scrs)
+	assert.Equal(t, 1, len(finalSCRs))
+	assert.Equal(t, 1, len(logs))
+}
+
+func TestProcessGetOriginalTxHashForRelayedIntraShard(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockSmartContractProcessorArguments()
+	arguments.ArgsParser = smartContract.NewArgumentParser()
+	shardCoordinator, _ := sharding.NewMultiShardCoordinator(2, 0)
+	arguments.ShardCoordinator = shardCoordinator
+	sc, _ := NewSmartContractProcessorV2(arguments)
+
+	scr := &smartContractResult.SmartContractResult{Value: big.NewInt(1), SndAddr: bytes.Repeat([]byte{1}, 32)}
+	scrHash := []byte("hash")
+
+	logHash := sc.getOriginalTxHashIfIntraShardRelayedSCR(scr, scrHash)
+	assert.Equal(t, scrHash, logHash)
+
+	scr.OriginalTxHash = []byte("originalHash")
+	scr.RelayerAddr = bytes.Repeat([]byte{1}, 32)
+	scr.SndAddr = bytes.Repeat([]byte{1}, 32)
+	scr.RcvAddr = bytes.Repeat([]byte{1}, 32)
+	logHash = sc.getOriginalTxHashIfIntraShardRelayedSCR(scr, scrHash)
+	assert.Equal(t, scr.OriginalTxHash, logHash)
+
+	scr.RcvAddr = bytes.Repeat([]byte{2}, 32)
+	logHash = sc.getOriginalTxHashIfIntraShardRelayedSCR(scr, scrHash)
+	assert.Equal(t, scrHash, logHash)
+}
+
+func TestProcess_createCompletedTxEvent(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockSmartContractProcessorArguments()
+	arguments.ArgsParser = smartContract.NewArgumentParser()
+	shardCoordinator, _ := sharding.NewMultiShardCoordinator(2, 0)
+	arguments.ShardCoordinator = shardCoordinator
+	completedLogSaved := false
+	arguments.TxLogsProcessor = &mock.TxLogsProcessorStub{SaveLogCalled: func(txHash []byte, tx data.TransactionHandler, vmLogs []*vmcommon.LogEntry) error {
+		for _, log := range vmLogs {
```

Comment:
Please rename this `log` to `vmLog`. Also on the next line. Will remove the weak warning

Reply:
done